File: app/src/main/kt/estyle/teabaike/dagger/component/TeaBaikeComponent.java
package estyle.teabaike.dagger.component;

import javax.inject.Singleton;

import dagger.Component;
import estyle.teabaike.activity.CollectionActivity;
import estyle.teabaike.activity.ContentActivity;
import estyle.teabaike.activity.SearchActivity;
import estyle.teabaike.activity.SplashActivity;
import estyle.teabaike.adapter.CollectionAdapter;
import estyle.teabaike.dagger.module.DataModule;
import estyle.teabaike.dagger.module.TimerModule;
import estyle.teabaike.fragment.MainFragment;
import estyle.teabaike.widget.HeadlineHeaderView;

@Singleton
@Component(modules = {DataModule.class, TimerModule.class})
public interface TeaBaikeComponent {

    void inject(MainFragment fragment);

    void inject(HeadlineHeaderView view);

    void inject(ContentActivity activity);

    void inject(SearchActivity activity);

    void inject(CollectionActivity activity);

    void inject(CollectionAdapter adapter);

    void inject(SplashActivity activity);

}

File: app/src/main/kt/estyle/teabaike/dagger/component/TeaBaikeComponent.kt
package estyle.teabaike.dagger.component

import dagger.Component
import estyle.teabaike.dagger.module.DataModule
import estyle.teabaike.viewmodel.CollectionViewModel
import estyle.teabaike.viewmodel.ContentViewModel
import javax.inject.Singleton

@Singleton
@Component(modules = [DataModule::class])
interface TeaBaikeComponent {

//    fun inject(dataSource: MainListDataSource)
//    fun inject(viewModel: MainViewModel)

    fun inject(viewModel: ContentViewModel)

//    fun inject(dataSource: SearchListDataSource)

    fun inject(viewModel: CollectionViewModel)

//    fun inject(activity: SplashActivity)

//    fun inject(viewModel: FeedbackViewModel)
}
